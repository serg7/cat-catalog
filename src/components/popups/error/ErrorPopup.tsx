import React, { useEffect, useState } from 'react';
import classNames from 'classnames';
import '../CommonPopup.css';

export const ErrorPopup = (props: { isShown: boolean }): JSX.Element => {
    const { isShown } = props;
    const [show, setIsShow] = useState(isShown);
    useEffect(() => {
       setIsShow(isShown);
    }, [props]);

    return (
        <div className={classNames('popup', { show })}>
          <span className="close" onClick={() => setIsShow(false)}>&times;</span>
           Network Error...
        </div>
    )
};

export default ErrorPopup;
