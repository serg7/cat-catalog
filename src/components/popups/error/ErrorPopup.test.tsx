import React from 'react';
import '@testing-library/jest-dom/extend-expect'
import { toBeInTheDocument, toHaveClass } from '@testing-library/jest-dom';
import { render, screen } from '@testing-library/react';
import ErrorPopup from './ErrorPopup';

test('should render Error popup', () => {
  render(<ErrorPopup isShown={true} message="test" />);
  const popup = screen.getByTestId('error-popup');

  expect(popup).toMatchSnapshot();
  expect(popup).toBeInTheDocument();
});