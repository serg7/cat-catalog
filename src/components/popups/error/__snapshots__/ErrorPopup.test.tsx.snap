// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render Detail popup 1`] = `
<div
  class="popup show"
  data-testid="error-popup"
>
  <span
    class="close"
  >
    ×
  </span>
  <div>
    test
  </div>
</div>
`;

exports[`should render Error popup 1`] = `
<div
  class="popup show"
  data-testid="error-popup"
>
  <span
    class="close"
  >
    ×
  </span>
  <div>
    test
  </div>
</div>
`;
